autode/wrappers/QChem.py

```diff
@@ -0,0 +1,502 @@
+
+            self.write('$molecule\n'
+                       f'{molecule.charge} {molecule.mult}')
+
+            for atom in molecule.atoms:
+                x, y, z = atom.coord
+                self.write(f'{atom.label:<3} {x:^12.8f} {y:^12.8f} {z:^12.8f}')
+
+            self.write('$end\n')
+            return None
+
+        def add_rem_block(self, calc) -> None:
+            """Add the $rem block"""
+            keywords = calc.input.keywords
+
+            if any('$' in word.lower() for word in keywords):
+                raise NotImplementedError('Cannot add $rem block - already '
+                                          'present in {keywords}')
+
+            self.write('$rem')
+
+            if calc.molecule.n_atoms > 1:         # Defaults to a single point
+                self._write_job_type(keywords)
+
+            self._write_keywords(keywords, molecule=calc.molecule)
+
+            if not isinstance(calc.input.keywords, kws.SinglePointKeywords):
+                self.write('symmetry False')
+                self.write('sym_ignore True')
+
+            self.write('$end\n')
+
+            return None
+
+        def _write_ecp(self, ecp_kwd, molecule) -> None:
+            """Write the effective core potential (ECP) block, if required"""
+
+            ecp_elems = set(atom.label for atom in molecule.atoms
+                            if atom.atomic_number >= ecp_kwd.min_atomic_number)
+
+            if len(ecp_elems) > 0:
+                logger.info(f'Writing ECP block for atoms {ecp_elems}')
+                self.write(f'ecp {ecp_kwd.qchem}')
+
+            return None
+
+        def _write_keywords(self, keywords, molecule) -> None:
+
+            for word in keywords:
+
+                if isinstance(word, kws.BasisSet):
+                    self.write(f'basis {word.qchem}')
+
+                elif isinstance(word, kws.Functional):
+                    self.write(f'method {word.qchem}')
+
+                elif isinstance(word, kws.DispersionCorrection):
+                    self.write(f'dft_d {word.qchem}')
+
+                elif isinstance(word, kws.MaxOptCycles):
+                    self.write(f'geom_opt_max_cycles {word}')
+
+                elif isinstance(word, kws.ECP):
+                    self._write_ecp(word, molecule=molecule)
+
+                elif isinstance(word, kws.ImplicitSolventType):
+
+                    if word.lower() != 'smd':
+                        err = f'Only SMD solvent is supported. Had: {word}'
+                        raise UnsuppportedCalculationInput(err)
+
+                    self.write('solvent_method smd')
+
+                elif 'jobtype' in word.lower():
+                    if molecule.n_atoms == 1 and 'opt' in word.lower():
+                        logger.warning('Cannot optimise a single atom')
+
+                    elif ' ts' in word.lower():
+                        logger.warning('TS optimisation - expected a completed'
```

Comment:
This warning makes it sound like something won't work, is that the case? (I think you've just made it so the hessian will also be calculated?)

Reply:
It will work fine – agree the warning is confusing (written before [this)](https://github.com/duartegroup/autodE/blob/ac4ececfcdba1a4d001018639f479196ea4a0fcb/autode/wrappers/QChem.py#L52). Will modify